open-api/lang-parser/src/main/java/io/deephaven/lang/parse/CompletionParser.java

```diff
@@ -27,26 +29,26 @@ public ParsedDocument parse(String document) throws ParseException {
     }
 
     @Override
-    public void open(final String text, final String uri, final String version, final Logger log) {
-        log.info()
+    public void open(final String text, final String uri, final String version) {
+        LOGGER.trace()
                 .append("Opening document ")
                 .append(uri)
                 .append("[")
                 .append(version)
                 .append("] ->\n")
                 .append(text)
                 .endl();
-        startParse(uri, log)
+        startParse(uri)
                 .requestParse(String.valueOf(version), text, false);
     }
 
-    private PendingParse startParse(String uri, Logger log) {
-        return docs.computeIfAbsent(uri, k -> new PendingParse(uri, log));
+    private PendingParse startParse(String uri) {
+        return docs.computeIfAbsent(uri, k -> new PendingParse(uri, LOGGER));
     }
 
     @Override
-    public void update(final String uri, final String version, final List<ChangeDocumentRequest.TextDocumentContentChangeEvent> changes, final Logger log) {
-        log.info()
+    public void update(final String uri, final String version, final List<ChangeDocumentRequest.TextDocumentContentChangeEvent> changes) {
+        LOGGER.trace()
```

Comment:
If you care about allocation (and I think you should), it would be worthwhile to enclose this and  some of the other trace calls in an `io.deephaven.io.logger.LoggerImpl#isTraceEnabled` test. Any of the logs that tries to allocate big strings should be guarded in this way.

Reply:
ah, makes sense.  Will do.

Reply:
A bit more specifically, the *bad* part is `.append(docs.keySet().toString())`, `.append(changes.toString())`.

The best way to do this imo is:

```
<T> LogEntry append(ObjFormatter<T> objFormatter, T t);
...
log...append(STRING_COLLECTION_FORMATTER, docs.keySet())
```
or something similar.